File: src/main/java/com/rss/config/AuthConfig.java
package com.rss.config;

import com.rss.config.properties.AuthProperties;
import org.apache.http.HttpStatus;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.HandlerInterceptor;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

@Configuration
public class AuthConfig implements HandlerInterceptor, WebMvcConfigurer {

    @Autowired
    private AuthProperties authProperties;

    @Override
    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
        if (request.getServletPath().equals("/error")) return true;

        String token = request.getHeader("Authorization");
        String ipAddress = request.getHeader("X-FORWARDED-FOR");
        if (ipAddress == null || ipAddress.isEmpty()) {
            ipAddress = request.getRemoteAddr();
        }

        if (authProperties.getClientIps() != null && authProperties.getClientIps().size() > 0) {
            if (!authProperties.getClientIps().contains(ipAddress)) {
                // IP not in whitelist
                response.setStatus(HttpStatus.SC_UNAUTHORIZED);
                System.out.println("Ip not in whitelist");
                return false;
            }
        }

        if (token == null) {
            response.setStatus(HttpStatus.SC_UNAUTHORIZED);
            System.out.println("Token is null");
            return false;
        }

        if (!token.equals(authProperties.getClientToken())) {
            response.setStatus(HttpStatus.SC_FORBIDDEN);
            System.out.println("Invalid Token");
            return false;
        }
        return true;
    }

    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        registry.addInterceptor(this);
    }
}

File: src/main/java/com/rss/config/AuthConfig.kt
package com.rss.config

import com.rss.config.properties.AuthProperties
import org.apache.http.HttpStatus
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.HandlerInterceptor
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@Configuration
open class AuthConfig : HandlerInterceptor, WebMvcConfigurer {
    @Autowired
    private val authProperties: AuthProperties? = null

    override fun preHandle(request: HttpServletRequest, response: HttpServletResponse, handler: Any): Boolean {
        if (request.servletPath == "/error") return true
        val token = request.getHeader("Authorization")
        var ipAddress = request.getHeader("X-FORWARDED-FOR")
        if (ipAddress == null || ipAddress.isEmpty()) {
            ipAddress = request.remoteAddr
        }
        if (authProperties!!.clientIps != null && authProperties.clientIps!!.size > 0) {
            if (!authProperties.clientIps!!.contains(ipAddress)) { // IP not in whitelist
                response.status = HttpStatus.SC_UNAUTHORIZED
                println("Ip not in whitelist")
                return false
            }
        }
        if (token == null) {
            response.status = HttpStatus.SC_UNAUTHORIZED
            println("Token is null")
            return false
        }
        if (token != authProperties.clientToken) {
            response.status = HttpStatus.SC_FORBIDDEN
            println("Invalid Token")
            return false
        }
        return true
    }

    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(this)
    }
}